code/_core/obj/item/fishing/rod.dm

```diff
@@ -180,16 +189,19 @@
 
 	INTERACT_CHECK
 
-	if(compact)
-		caller.to_chat(span("notice","Extend your fishing rod first!"))
-		return TRUE
 	if(object.plane >= PLANE_HUD)
 		return ..()
 
 	if(fishing_turf)
 		if(snagged_fish) //Caught something!
+			var/mob/living/C = caller // here comes the luck calc
+			var/lucktotal = clamp((C.get_attribute_level(ATTRIBUTE_LUCK) + (src.luck/2)- 75),1,75)	//base player+rod is 0 unless boosted, item luck affects it by half
```

Comment:
This function already exists in its own form; ``luck()``. check out ``/proc/luck()`` on how this works.

Reply:
should be fine now possibly, take a look at the changes whenever